kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,606 @@
+      return
+    fi
+  fi
+  eval $__started="false"
+}
+
+#
+# Function to check if server should start based on policy and current replicas
+# $1 - Current number of replicas
+# $2 - Server start policy
+# $3 - Replica count
+# $4 - Return value
+#
+function shouldStart {
+  local currentReplicas=$1
+  local policy=$2
+  local replicaCount=$3 
+  local __result=$4
+
+  if [ "$policy" == "ALWAYS" ]; then
+    eval $__result=True
+  elif [ "$policy" == "NEVER" ]; then
+    eval $__result=False
+  elif [ "${currentReplicas}" -lt "${replicaCount}" ]; then
+    eval $__result=True
+  else 
+    eval $__result=False
+  fi
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operatation string indicating whether to increment or decrement count
+# $4 - Return value containing replica update patch string
+# $5 - Retrun value containing updated replica count
+#
+function createReplicaPatch {
+  local domainJson=$1
+  local clusterName=$2
+  local operation=$3
+  local __result=$4
+  local __replicaCount=$5
+  local maxReplicas=""
+  local errorMessage="@@ ERROR: Maximum number of servers allowed (maxReplica = ${maxReplicas}) \
+are already running. Please increase cluster size to start new servers."
+
+  replicasCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).replicas"
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  replica=$(echo ${domainJson} | jq "${replicasCmd}")
+  if [[ -z "${replica}" || "${replica}" == "null" ]]; then
+    replica=$(echo ${domainJson} | jq .spec.replicas)
+  fi
+  if [ "${operation}" == "DECREMENT" ]; then
+    replica=$((replica-1))
+    if [ ${replica} -lt 0 ]; then
+      replica=0
+    fi
+  elif [ "${operation}" == "INCREMENT" ]; then
+    replica=$((replica+1))
+    maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+    if [ ${replica} -gt ${maxReplicas} ]; then
+      echo "${errorMessage}"
+      eval $__result="MAX_REPLICA_COUNT_EXCEEDED"
+      return
+    fi
+  fi
+
+  cmd="(.spec.clusters[] | select (.clusterName == \"${clusterName}\") \
+    | .replicas) |= ${replica}"
+  replicaPatch=$(echo ${domainJson} | jq "${cmd}" | jq -cr '(.spec.clusters)')
+  eval $__result="'${replicaPatch}'"
+  eval $__replicaCount="'${replica}'"
+}
+
+#
+# Function to validate whether a server belongs to a  cluster or is an independent managed server
+# $1 - Domain unique id.
+# $2 - Domain namespace.
+# $3 - Server name.
+# $4 - Return value indicating if server is valid (i.e. if it's part of a cluster or independent server).
+# $5 - Retrun value containting cluster name to which this server belongs.
+#
+function validateServerAndFindCluster {
+  local domainUid=$1
+  local domainNamespace=$2 
+  local serverName=$3
+  local __isValidServer=$4
+  local __clusterName=$5
+  local serverCount=""
```

Comment:
It's more clear if script set variables on failure too, so at start could do something like:
eval $__isValidServer=false
eval $__clusterName=UNKNOWN

Reply:
Changed.